apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -124,6 +130,38 @@ const aichatSlice = createSlice({
         chatMessage.status = status;
       }
     },
+    setAiCustomizations: (state, action: PayloadAction<AiCustomizations>) => {
+      state.currentAiCustomizations = action.payload;
+    },
+    setAiCustomizationProperty: (
+      state,
+      action: PayloadAction<{
+        property: keyof AiCustomizations;
+        value: AiCustomizations[typeof property];
+      }>
+    ) => {
+      const {property, value} = action.payload;
+      const updatedAiCustomizations: AiCustomizations = {
+        ...state.currentAiCustomizations,
+        [property]: value,
+      };
+      state.currentAiCustomizations = updatedAiCustomizations;
```

Comment:
nit: I think you may be able to just do `state.currentAiCustomizations[property] = value` here?

Reply:
Yeah I had that originally, TS doesn't like it?

```
ERROR in ./src/aichat/redux/aichatRedux.ts:143:7
TS2322: Type 'string | number | string[] | ModelCardInfo' is not assignable to type 'never'.
  Type 'string' is not assignable to type 'never'.
    141 |     ) => {
    142 |       const {property, value} = action.payload;
  > 143 |       state.currentAiCustomizations[property] = value;
        |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    144 |     },
    145 |     setModelCardProperty: (
    146 |       state,

Found 1 error in 4952 ms.

Reply:
Oh huh interesting. Not sure why it's considering that type `never`, maybe something with the `WriteableDraft` type that redux toolkit creates? But anyway what you have is fine in that case!